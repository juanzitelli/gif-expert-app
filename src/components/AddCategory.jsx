import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

	const [inputValue, setInputValue] = useState("");

	const handleInputChange = ({ target }) => setInputValue(target.value)

	const handleSubmit = (event) => {
		event.preventDefault()
		inputValue && setCategories(prevCategories => [inputValue, ...prevCategories])
		setInputValue('')
		console.log("Submitted!")
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				name="category"
				id="category"
				value={inputValue}
				onChange={handleInputChange}
			/>
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}

export default AddCategory
