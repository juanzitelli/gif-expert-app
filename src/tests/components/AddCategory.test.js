import React from "react";
import "@testing-library/jest-dom";
import AddCategory from "../../components/AddCategory";
import { shallow } from "enzyme";
describe("Pruebas en componente AddCategory", () => {
  const setCategories = jest.fn();
  let wrapper;
  const value = "Hola mundo";
  const event = {
    target: {
      value,
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("AddCategoy se renderiza correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", event);
  });
  test("NO debe postear la informaciÃ³n con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test("Debe ser llamado el setCategories Y limpiar la caja de texto ", () => {
    wrapper.find("input").simulate("change", event);
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find("input").prop('value')).toBe('');
})});
