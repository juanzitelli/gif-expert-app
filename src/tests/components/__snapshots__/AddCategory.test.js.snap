// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AddCategory AddCategoy se renderiza correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="category"
    name="category"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
